pyro/infer/mcmc/hmc.py

```diff
@@ -210,7 +210,8 @@ def sample(self, trace):
             z = z_new
 
         if self.adapt_step_size:
-            accept_prob = (-delta_energy).exp().clamp(max=1).item()
+            accept_prob = (-delta_energy).exp().clamp(max=1).item() if not torch_isnan(delta_energy) \
+                else torch.tensor(0.0)
```

Comment:
Just a note if we put comments related to `nan`: I think that we should also check the logic of `direction` in `_find_reasonable_step_size`. When encounter `nan`, we should reduce step_size (which is how the current behavior works). So it would be good if we put a comment there too.

Reply:
I think that logic should be fine - it will set direction = -1 if we encounter `NaN` which is what we want:
```
direction = 1 if target_accept_logprob < -delta_energy else -1
```

Will add a comment there.

Reply:
Yes, that logic is fine. I just wanted to mention about some relevant issues, which adding comments is especially helpful.